Modules/_json.c

```diff
@@ -1491,17 +1492,73 @@ encoder_listencode_obj(PyEncoderObject *s, _PyUnicodeWriter *writer,
     }
 }
 
+static int
+encoder_encode_key_value(PyEncoderObject *s, _PyUnicodeWriter *writer, bool *first,
```

Comment:
This is the original loop body, I renamed `kstr` to `keystr` and use bool instead of a counter to not add a comma for the first element.